collect_app/src/main/java/org/odk/collect/android/application/initialization/ApplicationInitializer.java

```diff
@@ -0,0 +1,119 @@
+package org.odk.collect.android.application.initialization;
+
+import android.app.Application;
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatDelegate;
+
+import com.crashlytics.android.Crashlytics;
+import com.evernote.android.job.JobManager;
+import com.evernote.android.job.JobManagerCreateException;
+
+import net.danlew.android.joda.JodaTimeAndroid;
+
+import org.odk.collect.android.BuildConfig;
+import org.odk.collect.android.application.Collect;
+import org.odk.collect.android.geo.MapboxUtils;
+import org.odk.collect.android.jobs.CollectJobCreator;
+import org.odk.collect.android.preferences.AdminSharedPreferences;
+import org.odk.collect.android.preferences.AutoSendPreferenceMigrator;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+import org.odk.collect.android.utilities.LocaleHelper;
+import org.odk.collect.android.utilities.NotificationUtils;
+import org.odk.collect.utilities.UserAgentProvider;
+
+import java.util.Locale;
+
+import timber.log.Timber;
+
+public class ApplicationInitializer {
+
+    private final Application context;
+    private final CollectJobCreator collectJobCreator;
+    private final SharedPreferences metaSharedPreferences;
+    private final UserAgentProvider userAgentProvider;
+    private final GeneralSharedPreferences generalSharedPreferences;
+    private final AdminSharedPreferences adminSharedPreferences;
+
+    public ApplicationInitializer(Application context, CollectJobCreator collectJobCreator, SharedPreferences metaSharedPreferences, UserAgentProvider userAgentProvider) {
+        this.context = context;
+        this.collectJobCreator = collectJobCreator;
+        this.metaSharedPreferences = metaSharedPreferences;
+        this.userAgentProvider = userAgentProvider;
+
+        generalSharedPreferences = GeneralSharedPreferences.getInstance();
+        adminSharedPreferences = AdminSharedPreferences.getInstance();
+    }
+
+    public void initializePreferences() {
+        performMigrations();
+        reloadSharedPreferences();
+    }
+
+    public void initializeFrameworks() {
+        NotificationUtils.createNotificationChannel(context);
+        initializeJobManager();
+        JodaTimeAndroid.init(context);
+        initializeLogging();
+        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
+
+        new com.google.android.gms.maps.MapView(context).onCreate(null);
```

Comment:
Is there any reason you unpacked this? I mean having it in a separate method like `initializeMaps` or `initializeMapProviders` seems more elegant.

Reply:
It felt fine to have them in the `initializeFrameworks` to me but I don't have any strong feelings on it. I'll put the method back but call it from the same place.